rows.go

```diff
@@ -64,12 +65,24 @@ func (rows *mysqlRows) Columns() []string {
 	return columns
 }
 
+func (rows *mysqlRows) setFinish(f func()) {
```

Comment:
`mc.startWatch` and `mc.finish()` must be paired, but `mysqlRows` is used internally.

- getWarnings: https://github.com/go-sql-driver/mysql/blob/v1.3/errors.go#L92

So, I use`setfinish` to prevent `mc.finish()` from being called unexpectedly.